File: golang/693.binary-number-with-alternating-bits.go
func hasAlternatingBits(n int) bool {
  prev := 1 ^ n & 1
  
  for ; n > 0; n >>= 1 {
    if n & 1 == prev {
      return false
    }
    
    prev = n & 1
  }

  return true
}

File: cpp/693.binary-number-with-alternating-bits.cpp
class Solution {
public:
    bool hasAlternatingBits(int n) {
        int prev = 1 ^ n & 1;

        for (int x = n & 1; n; n >>= 1, x = n & 1) {
            if (prev == x) {
                return false;
            }

            prev = x;
        }

        return true;
    }
};